prymer/primer3/primer3.py

```diff
@@ -399,54 +380,38 @@ def design(self, design_input: Primer3Input) -> Primer3Result:  # noqa: C901
             **global_primer3_params,
             **design_input.to_input_tags(design_region=design_region),
         }
-        # Submit inputs to primer3
-        for tag, value in assembled_primer3_tags.items():
-            self._subprocess.stdin.write(f"{tag}={value}")
-            self._subprocess.stdin.write("\n")
-        self._subprocess.stdin.write("=\n")
-        self._subprocess.stdin.flush()
 
-        error_lines: list[str] = []  # list of errors as reported by primer3
-        primer3_results: dict[str, str] = {}  # key-value pairs of results reported by Primer3
+        # split the tags into sequence and non-sequence tags
+        seq_args = {}
+        global_args = {}
+        for tag, value in assembled_primer3_tags.items():
+            tag_str = f"{tag}"
+            if tag_str.startswith("SEQUENCE"):
+                seq_args[tag_str] = value
+            else:
+                global_args[tag] = value
+        design_primers_retval = primer3.design_primers(seq_args=seq_args, global_args=global_args)
+        primer3_results: dict[str, Any] = {}  # key-value pairs of results reported by Primer3
+
+        for key, value in design_primers_retval.items():
+            # Because Primer3 will emit both the input given and the output generated, we
+            # discard the input that is echo'ed back by looking for tags (keys)
+            # that do not match any Primer3InputTag
+            if not any(key == item.value for item in Primer3InputTag):
+                primer3_results[key] = value
```

Comment:
Is this actually needed?  I don't think there's any harm in allow the inputs to be present in the dict() used below.  

If you feel strongly about this .. perhaps we have a more efficient way than looping through all the values in Primer3InputTag - e.g. create a _ALL_INPUT_TAG_STRINGS set and at the class level and check for membership?

Reply:
I'll keep removing them, we can open a PR later if you want to see them.  I made it not inefficient via set membership.